index.d.ts

```diff
@@ -0,0 +1,38 @@
+import * as Http from 'http';
+import * as Https from 'https';
+import { Stream } from 'pump';
+import * as Undici from 'undici';
+
+interface Options {
+  base?: string;
+  http2?: boolean;
+  undici?: Undici.Pool.Options;
+  cacheURLs?: number;
+  requests?: {
+    http?: Http.Agent,
+    https?: Https.Agent
+  };
+  keepAliveMsecs?: number;
+  maxSockets?: number;
+  rejectUnauthorized?: boolean;
+}
+
+
+declare function fastProxy(options?: Options): {
```

Comment:
Hi @mcollina, while I am fine with this changes, I wanted to double check with you regarding naming conventions across the fastify org. Would you agree on using `fastProxy` as function name?